Snowflake.Data.Tests/IntegrationTests/SFConnectionIT.cs

```diff
@@ -1540,6 +1540,51 @@ public void TestInvalidProxySettingFromConnectionString()
             }
         }
 
+        [Test]
+        [TestCase("*")]
+        [TestCase("*{0}*")]
+        [TestCase("^*{0}*")]
+        [TestCase("*{0}*$")]
+        [TestCase("^*{0}*$")]
+        [TestCase("^nonmatch*{0}$|*")]
+        [TestCase("*a*", "a")]
+        [TestCase("*la*", "la")]
+        public void TestNonProxyHostShouldBypassProxyServer(string regexHost, string proxyHost = null)
+        {
+            using (var conn = new SnowflakeDbConnection())
+            {
+                var host = ResolveHostToUseFromTestConfig(ConnectionString);
+                var nonProxyHosts = string.Format(regexHost, $"{host}");
+                var proxyHostForConnection = proxyHost ?? "proxyserverhost";
+                conn.ConnectionString =
+                    $"{ConnectionString}USEPROXY=true;PROXYHOST={proxyHostForConnection};NONPROXYHOSTS={nonProxyHosts};PROXYPORT=3128;";
+                conn.Open();
+                Assert.AreEqual(ConnectionState.Open, conn.State);
+            }
+        }
+
+        [Test]
+        [TestCase("invalid{0}")]
+        [TestCase("*invalid{0}*")]
+        [TestCase("^invalid{0}$")]
+        [TestCase("*a.b")]
+        [TestCase("a", "a")]
+        [TestCase("la", "la")]
+        public void TestNonProxyHostShouldNotBypassProxyServer(string regexHost, string proxyHost = null)
+        {
+            using (var conn = new SnowflakeDbConnection())
+            {
+                var nonProxyHosts = string.Format(regexHost, $"{testConfig.host}");
+                var proxyHostForConnection = proxyHost ?? "proxyserverhost";
+                conn.ConnectionString =
+                    $"{ConnectionString}connection_timeout=5;USEPROXY=true;PROXYHOST={proxyHostForConnection};NONPROXYHOSTS={nonProxyHosts};PROXYPORT=3128;";
+                var exception = Assert.Throws<SnowflakeDbException>(() => conn.Open());
```

Comment:
let's write a comment how exception on connecting proves that we didn't bypass the proxy server because it could be not obvious to the reader. Also verify if it is possible to write a unit test instead of integration tests, because integration tests are much slower.

Reply:
Added comment to specify how the test is handled. This is an integration test because we are validating the external WebProxy which has the functionality to determine if the proxy server is bypassed or not.

Reply:
you wrote: "If the nonproxyhosts property fails to bypass the server, the proxy will be use but it will return a exception because is not configure in the testing environment so the exception is used to verify that was not bypass."
It contains the reason WHY - that is good. But maybe we could use the shorter form as I suggested in the previous case: "The connection would fail to open if the web proxy would be used because the proxy is configured to a non-existent host"

Reply:
Done